python/testSrc/com/jetbrains/python/PyTypeTest.java

```diff
@@ -816,6 +816,12 @@ public void testDefaultTupleParameterMember() {
            "  expr, foo = xs\n");
   }
 
+  // PY-19826
+  public void testListFromTupleType() {
+    doTest("List[Union[str, int]]",
+           "t = ('1',2,3); expr = list(t)");
+  }
+
```

Comment:
My idea was to treat the tuple as a single generic typed collection when accessed through PyCollectionType API and as multi-generic when accessed through PyTupleType API (unless homogeneous).  Does it even make sense to have PyTupleTyple implement PyCollectionType if you have to special-case it in PyTypeChecker? Maybe just scrap the idea. I'm not too attached to my code...